internal/pkg/agent/application/upgrade/marker_watcher.go

```diff
@@ -34,13 +37,18 @@ func newMarkerFileWatcher(upgradeMarkerFilePath string, logger *logger.Logger) M
 		markerFilePath: upgradeMarkerFilePath,
 		logger:         logger,
 		updateCh:       make(chan UpdateMarker),
+		upgradeStarted: false,
 	}
 }
 
 func (mfw *MarkerFileWatcher) Watch() <-chan UpdateMarker {
 	return mfw.updateCh
 }
 
+func (mfw *MarkerFileWatcher) SetUpgradeStarted() {
+	mfw.upgradeStarted = true
```

Comment:
even if setting a boolean is probably atomic according to Go's memory model we may want to use a mutex or an atomic boolean here for correctness ?